### PR TITLE
Implement basic e2e test for 2 players

### PR DESCRIPTION
**Pull request by zablaveki**
Refactor the page objects: add BasePageElement and BasePage classes
Remove AdminPage: use BatchesPage directly

### CODE REVIEW

File: e2e-tests/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+## How to run the tests
+
+1. Install dependencies
+
+```npm install```
+
+2. run tests
+
+```npm run test:e2e`
```
Comment (npaton) [inline e2e-tests/README.md:16]: Broken code tag.

File: e2e-tests/page-objects/BasePage.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    private context: BrowserContext;
+
+    constructor({ browser, baseUrl }: BasePageConstructor) {
+        this.baseUrl = baseUrl ?? '';
+        this.browser = browser;
+    }
+
+    protected async initContext() {
+        this.context = await this.browser.newContext();
+        this.page = await this.context.newPage();
```
Comment (npaton) [inline e2e-tests/page-objects/BasePage.ts:27]: Should this only ever run once per page? Should we guard double executions with a flag?

**Review by npaton — APPROVED** [review]
Looking good 🙌

File: e2e-tests/README.md (hunk 1 above)
Comment (zablaveki) [reply e2e-tests/README.md:16]: Good catch!

File: e2e-tests/page-objects/BasePage.ts (hunk 2 above)
Comment (zablaveki) [reply e2e-tests/page-objects/BasePage.ts:27]: Yes, makes sense. Initially, I've added a separate init method that would create the page context, but I forgot to call it a few times (and got undefined in runtime :D) , so having the context creation in the open method was reasonable. Will improve in the next PR :)

**Pull request merged**